libs/pika/runtime_configuration/src/runtime_configuration.cpp

```diff
@@ -42,7 +44,154 @@
+        if (argv0_.length() > 0)
+        {
+            // Check for an absolute path.
+            if (argv0_[0] == '/') return argv0_;
+
+            // Check for a relative path.
+            if (argv0_.find('/') != std::string::npos)
+            {
+                // Get the current working directory.
+
+                // NOTE: getcwd does give you a null terminated string,
+                // while readlink (above) does not.
+                if (::getcwd(buf, PATH_MAX))
+                {
+                    r = buf;
+                    r += '/';
+                    r += argv0_;
+                    return r;
+                }
+            }
+
+            // Search PATH
+            char const* epath = ::getenv("PATH");
+            if (epath)
+            {
+                std::vector<std::string> path_dirs;
+
+                pika::detail::split(path_dirs, epath, pika::detail::is_any_of(":"),
+                    pika::detail::token_compress_mode::on);
+
+                for (std::uint64_t i = 0; i < path_dirs.size(); ++i)
+                {
+                    r = path_dirs[i];
+                    r += '/';
+                    r += argv0_;
+
+                    // Can't use Boost.Filesystem as it doesn't let me access
+                    // st_uid and st_gid.
+                    struct stat s;
+
+                    // Make sure the file is executable and shares our
+                    // effective uid and gid.
+                    // NOTE: If someone was using a pika application that was
+                    // seteuid'd to root, this may fail.
+                    if (0 == ::stat(r.c_str(), &s))
+                        if ((s.st_uid == ::geteuid()) && (s.st_mode & S_IXUSR) &&
+                            (s.st_gid == ::getegid()) && (s.st_mode & S_IXGRP) &&
+                            (s.st_mode & S_IXOTH))
+                            return r;
+                }
+            }
+        }
+
+        PIKA_THROW_EXCEPTION(pika::error::dynamic_link_failure, "get_executable_filename",
+            "unable to find executable filename");
+
+#elif defined(__APPLE__)
+        PIKA_UNUSED(argv0);
+
+        char exe_path[PATH_MAX + 1];
+        std::uint32_t len = sizeof(exe_path) / sizeof(exe_path[0]);
+
+        if (0 != _NSGetExecutablePath(exe_path, &len))
+        {
+            PIKA_THROW_EXCEPTION(pika::error::dynamic_link_failure, "get_executable_filename",
+                "unable to find executable filename");
+        }
+        exe_path[len - 1] = '\0';
+        r = exe_path;
+
+#elif defined(__FreeBSD__)
+        PIKA_UNUSED(argv0);
+
+        int mib[] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
+        size_t cb = 0;
+        sysctl(mib, 4, nullptr, &cb, nullptr, 0);
+        if (cb)
+        {
+            std::vector<char> buf(cb);
+            sysctl(mib, 4, &buf[0], &cb, nullptr, 0);
+            std::copy(&buf[0], &buf[cb], std::back_inserter(r));
```

Comment:
Make sure you include this change: https://github.com/pika-org/pika/pull/1171/files.